torch/nn/attention/_flex_attention.py

```diff
@@ -278,37 +338,65 @@ def _convert_block_mask_to_mask(
 
 
 def _create_sparse_block_from_block_mask(
-    block_mask: torch.Tensor,
+    block_mask: Tuple[torch.Tensor, Optional[torch.Tensor]],
+    mask_fn: Callable,
     KV_BLOCK_SIZE: int = _DEFAULT_SPARSE_BLOCK_SIZE,
     Q_BLOCK_SIZE: int = _DEFAULT_SPARSE_BLOCK_SIZE,
 ) -> BlockMask:
-    device = block_mask.device
-    block_mask = block_mask.to(dtype=torch.int8)
-    kv_num_blocks = block_mask.sum(dim=3)
-    kv_indices = torch.argsort(block_mask, dim=3, descending=True, stable=True)
-    q_num_blocks = block_mask.sum(dim=2)
-    q_indices = torch.argsort(block_mask, dim=2, descending=True, stable=True).permute(
-        0, 1, 3, 2
-    )
+    full_blocks, partial_blocks = block_mask
+
+    def create_sparse_block_from_block_mask_inner(block_mask) -> Tuple:
+        block_mask = block_mask.to(dtype=torch.int32)
+        kv_num_blocks = block_mask.sum(dim=3)
+        kv_indices = torch.argsort(block_mask, dim=3, descending=True, stable=True)
+        q_num_blocks = block_mask.sum(dim=2)
+        q_indices = torch.argsort(
+            block_mask, dim=2, descending=True, stable=True
+        ).permute(0, 1, 3, 2)
+        return (
+            kv_num_blocks.to(torch.int32).to(block_mask.device).contiguous(),
+            kv_indices.to(torch.int32).to(block_mask.device).contiguous(),
+            q_num_blocks.to(torch.int32).to(block_mask.device).contiguous(),
+            q_indices.to(torch.int32).to(block_mask.device).contiguous(),
+        )
+
+    full_bm = create_sparse_block_from_block_mask_inner(full_blocks)
+    if partial_blocks is not None:
+        partial_bm = create_sparse_block_from_block_mask_inner(partial_blocks)
+    else:
+        # Triton kernel would skip computation for these blocks.
+        z, h, m, k = full_blocks.shape
+        partial_bm = (
+            torch.zeros([1, 1, 1], dtype=torch.int32, device=full_blocks.device),
+            torch.zeros([1, 1, 1, 1], dtype=torch.int32, device=full_blocks.device),
+            torch.zeros([1, 1, 1], dtype=torch.int32, device=full_blocks.device),
+            torch.zeros([1, 1, 1, 1], dtype=torch.int32, device=full_blocks.device),
+        )
+
     return BlockMask(
-        kv_num_blocks=kv_num_blocks.to(torch.int32).to(device).contiguous(),
-        kv_indices=kv_indices.to(torch.int32).to(device).contiguous(),
-        q_num_blocks=q_num_blocks.to(torch.int32).to(device).contiguous(),
-        q_indices=q_indices.to(torch.int32).to(device).contiguous(),
-        KV_BLOCK_SIZE=KV_BLOCK_SIZE,
-        Q_BLOCK_SIZE=Q_BLOCK_SIZE,
+        full_bm[0],
```

Comment:
You can simplify these by splatting them out I think, like `*full_bm, *partial_bm`.

Reply:
Tried, but it seems lint can't recognize it well, lol